hbase-server/src/test/java/org/apache/hadoop/hbase/coprocessor/SimpleRegionObserver.java

```diff
@@ -685,6 +687,40 @@ public StoreFileReader postStoreFileReaderOpen(ObserverContext<RegionCoprocessor
     return reader;
   }
 
+  @Override
+  public void preStoreScannerOpen(ObserverContext<RegionCoprocessorEnvironment> ctx,
+    Store store, ScanOptions options) throws IOException {
+    if (options.getScan().getTimeRange().isAllTime()) {
```

Comment:
I think this class is only used to test whether a specific CP method has been called? What is this used for?

Reply:
See TestRegionCoprocessorHost. It's used to test that the scanner coproc hooks are wired up correctly and can alter a ScanInfo.

Reply:
OK, the rebuild of the ScanInfo is done in RegionCoprocessorHost, I used to thought it is done in HStore on some other places.